### PR TITLE
"🙏 Please Work" - Magee (PWM F4xx Support)

### PR DESCRIPTION
**Pull request by tmb5932**
Added support for PWM on f4xx.

### CODE REVIEW

File: samples/pwm/main.cpp [hunk 1]
```diff
@@ -13,7 +13,7 @@ int main() {
     // Initialize system
     EVT::core::platform::init();
 
-    IO::PWM& pwm = IO::getPWM<IO::Pin::PC_0>();
+    IO::PWM& pwm = IO::getPWM<IO::Pin::PC_8>();
```
Comment (mjh9585) [inline samples/pwm/main.cpp]: Change the pin to one that is supported across all of the chips.

File: src/io/platform/f4xx/PWMf4xx.cpp [hunk 2]
```diff
@@ -0,0 +1,213 @@
+#include <EVT/io/platform/f4xx/GPIOf4xx.hpp>
+#include <EVT/io/platform/f4xx/PWMf4xx.hpp>
+
+namespace EVT::core::IO {
+
+/**
+ * Get the TIM instance and channel associated with a pin. This information
+ * is pulled from the STM32F446re documentation.
```
Comment (mjh9585) [inline src/io/platform/f4xx/PWMf4xx.cpp]: Capitalize "RE" in the part number

**Review by mjh9585 — APPROVED** [review]
Looks good! Just a few minor changes.

File: docs/source/api/stm32f3xx/stm32f3xx_io.rst [hunk 3]
```diff
@@ -35,7 +35,7 @@ I2C
 PWM
 ===
 
-.. doxygenclass:: EVT::core::IO::PWMf3xx
+.. doxygenclass:: EVT::core::IO::PWMf4xx
```
Comment (mjmagee991) [inline docs/source/api/stm32f3xx/stm32f3xx_io.rst:38]:
```suggestion
.. doxygenclass:: EVT::core::IO::PWMf3xx
```

File: include/EVT/io/platform/f4xx/PWMf4xx.hpp [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+class PWMf4xx : public PWM {
+public:
+    /**
+     * Setup the given pin for PWM usage.
+     *
+     * @param pin[in] The pin to setup for PWM
+     */
+    PWMf4xx(Pin pin);
+
+    void setDutyCycle(uint32_t dutyCycle);
```
Comment (mjmagee991) [inline include/EVT/io/platform/f4xx/PWMf4xx.hpp:21]:
```suggestion
    void setDutyCycle(uint32_t dutyCycle) override;
```
Add this to all the public functions

File: samples/pwm/main.cpp
Comment (mjmagee991) [inline samples/pwm/main.cpp:1]: Revert your changes to this file

File: src/io/platform/f4xx/PWMf4xx.cpp [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    TIM_OC_InitTypeDef sConfigOC = {0};
+    sConfigOC.OCMode = TIM_OCMODE_PWM1;
+    sConfigOC.Pulse = dutyCycle * halTIM.Init.Period / 100;
+    sConfigOC.OCPolarity = TIM_OCPOLARITY_HIGH;
+    sConfigOC.OCNPolarity = TIM_OCNPOLARITY_HIGH;
+    sConfigOC.OCFastMode = TIM_OCFAST_DISABLE;
+    sConfigOC.OCIdleState = TIM_OCIDLESTATE_RESET;
+    sConfigOC.OCNIdleState = TIM_OCNIDLESTATE_RESET;
+
+    HAL_TIM_PWM_Stop(&halTIM, halTIMChannelID);
```
Comment (mjmagee991) [inline src/io/platform/f4xx/PWMf4xx.cpp:172]: Can you add this to PWMf3xx.cpp as well? Be sure to test that things still work after the change

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Looks great! A couple minor details to iron out

**Review by mjmagee991 — APPROVED** [review]
Nice work

**Pull request merged**